Misc/quick changes

MultidimensionalSock opened the pull request.

MultidimensionalSock committed:
Merge pull request #29 from Ethricus/develop

Develop

MultidimensionalSock committed:
level fix

MultidimensionalSock committed:
non-incline movement

MultidimensionalSock committed:
inclines and level fix for collisions on ramps

MultidimensionalSock committed:
change skybox

MultidimensionalSock committed:
sprint fix

MultidimensionalSock committed:
sprint fix

MultidimensionalSock committed:
Update PlayerControls.cs

Ethirix commented on Lares/Assets/Player/Scripts/PlayerAnimation.cs:
```diff
@@ -8,19 +8,33 @@ public class PlayerAnimation : MonoBehaviour
             _input = GetComponentInParent<PlayerInput>();
             _animator = GetComponent<Animator>();
-            _input.currentActionMap.FindAction("MovePlayer").performed += MoveStart;
-            _input.currentActionMap.FindAction("MovePlayer").canceled += MoveEnd;
+            _rigidbody = GetComponentInParent<Rigidbody>();
+            //_input.currentActionMap.FindAction("MovePlayer").performed += MoveStart;
+            //_input.currentActionMap.FindAction("MovePlayer").canceled += MoveEnd;
```
Shouldn't be commented out?
If so, why not removed?

Ethirix commented on Lares/Assets/Player/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,202 @@
+
+        private Vector3 _inclineHeight;
+
+        private bool _isGrounded
+        {
+            set
+            {
+                _isGroundedInternal = value;
+                if (!_isGroundedInternal)
+                    _movementForce *= _inAirMovementModifier;
+                else
+                    _movementForce /= _inAirMovementModifier;
+            }
+
+            get { return _isGroundedInternal; }
+        }
+
+        private bool _isGroundedInternal;
```
Can make IsGrounded public - it may be useful for external things, maybe stopping the player from doing an interaction (wouldn't be on PlayerController, in my opinion) if not on the ground?

To protect the value, you can make the setter private:
```suggestion
        public bool IsGrounded
        {
            private set
            {
                _isGrounded = value;
                if (!_isGrounded)
                    _movementForce *= _inAirMovementModifier;
                else
                    _movementForce /= _inAirMovementModifier;
            }

            get { return _isGrounded; }
        }

        private bool _isGrounded;
```

Ethirix commented on Lares/Assets/Player/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,202 @@
+
+        private bool _isGroundedInternal;
+
+        private int groundColliderCount;
```
```suggestion
        private int _groundColliderCount;
```

Ethirix commented on Lares/Assets/Player/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,202 @@
+        private bool _isSprinting;
+
+        [Header("Raycast Settings")]
+        [SerializeField] float raycastDistance;
```
```suggestion
        [SerializeField] float _raycastDistance;
```

Ethirix commented on Lares/Assets/Player/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,202 @@
+        private int groundColliderCount;
+        private bool _canJump => (_isGrounded && _jumpCount == 0) || (!_isGrounded && _jumpCount == 1);
+        private int _jumpCount = 0;
+        //private bool _isOnIncline => OnIncline();
```
Do you want this to be a value you can use?
If not, remove.

Ethirix commented on Lares/Assets/Player/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,202 @@
+            _input.currentActionMap.FindAction("MovePlayer").performed += OnPlayerStartMoving;
+            _input.currentActionMap.FindAction("MovePlayer").canceled += OnPlayerStopMoving;
+            _input.currentActionMap.FindAction("Jump").performed += Jump;
+            //    //_input.currentActionMap.FindAction("Interact").performed += Interact;
```
Could also add a TODO on this.

Ethirix commented on Lares/Assets/Player/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,202 @@
+
+        private IEnumerator Movement()
+        {
+            GameObject rotPoint = transform.GetChild(0).gameObject;
+            GameObject model = transform.GetChild(1).gameObject;
```
This is very prone to error if you happen to ever change the order of children - could make these set through `SerializedField`s.

Ethirix commented on Lares/Assets/Player/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,202 @@
+
+                float currentMovementForce = _movementForce;
+                float currentMaxSpeed = _maxSpeed;
+                if (_isSprinting) { currentMovementForce *= _sprintSpeedMultiplier; currentMaxSpeed *= _sprintSpeedMultiplier; }
```
For consistency: 
```suggestion
                if (_isSprinting)
                { 
                    currentMovementForce *= _sprintSpeedMultiplier; currentMaxSpeed *= _sprintSpeedMultiplier; 
                }
```

Ethirix commented on Lares/Assets/Player/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,202 @@
+        #region Collisions Checks
+        private void OnCollisionEnter(Collision collision)
+        {
+            if (collision.gameObject.CompareTag("Ground"))
+            {
+                groundColliderCount++;
+                _jumpCount = 0;
+                _isGrounded = true;
+            }
```
I don't know if this is simply temporary (would be nice to comment that it is, if so), however, this is prone to error.

For example: if you stood on a wall, you would not be marked as touching the ground - maybe worth swapping this to a very short raycast?

Ethirix commented on Lares/Assets/Player/Scripts/PlayerController.cs:
```diff
@@ -0,0 +1,202 @@
+
+        #endregion
+
+        private void LateUpdate()
+        {
+            //transform.GetChild(1).gameObject.transform.localPosition = Vector3.zero;
+            _rigidbody.useGravity = !OnIncline();
+        }
```
This is an odd choice for this I don't even know what to think of it.

Ethirix reviewed (changes requested):
Please review comments :)

MultidimensionalSock committed:
ground collision broken

collision with ground happens even when the ground isnt directly below you as long as the gameobject being collided with is tagged as "Ground".

MultidimensionalSock committed:
Merge branch 'misc/quick-changes' of https://github.com/ErithacusProject/Lares-Project into misc/quick-changes

# Conflicts:
#	Lares/Assets/Player/Scripts/PlayerController.cs
